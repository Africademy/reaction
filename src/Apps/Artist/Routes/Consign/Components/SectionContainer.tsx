import { Color, color, Flex, FlexProps } from "@artsy/palette"
import React from "react"

// Doesn't exist in design system
const lightPurple = "#EAE5E7"

interface SectionContainerProps extends FlexProps {
  background?: Color | typeof lightPurple
}

export const SectionContainer: React.FC<SectionContainerProps> = ({
  children,
  background = "white100",
  ...rest
}) => {
  // For sections with bg-colors outside of our design system
  const bg = background === lightPurple ? lightPurple : color(background)

  return (
    <Flex
      background={bg}
      flexDirection="column"
      justifyContent="center"
      alignItems="center"
      px={2}
      pt={6}
      pb={6}
      {...rest}
    >
      {children}
    </Flex>
  )
}
