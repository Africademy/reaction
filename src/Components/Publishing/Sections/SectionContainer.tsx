import styled from "styled-components"
import { pMedia } from "../../Helpers"
import { ArticleLayout, SectionData } from "../Typings"

const ColumnWidth = "680px"
const OverflowWidth = "780px"
const BlockquoteWidth = "900px"
const Fillwidth = "100%"

export const SectionContainer = styled.div.attrs<{
  section?: SectionData
  articleLayout?: ArticleLayout
}>({})`
  box-sizing: border-box;
  display: flex;
  margin: auto;
  margin-bottom: 40px;
  width: ${props => getSectionWidth(props.section, props.articleLayout)};
  max-width: 100%;

  ${props => pMedia.xl`
    ${props.articleLayout === "standard" &&
      `
      width: ${ColumnWidth}
    `}
  `} ${props => pMedia.md`
    padding: ${getSectionMobilePadding(props.section)};
  `};
`

export const getSectionWidth = (
  section?: SectionData,
  articleLayout?: ArticleLayout
) => {
  const layout = (section && section.layout) || "column_width"
  const maybeOverflow =
    layout === "overflow_fillwidth" ? OverflowWidth : ColumnWidth
  const isText = section && section.type === "text"
  const isBlockquote = isText && section.body.includes("<blockquote>")

  switch (articleLayout) {
    case "standard": {
      return maybeOverflow
    }
    case "feature": {
      if (isBlockquote) {
        return BlockquoteWidth
      } else if (layout === "fillwidth") {
        return Fillwidth
      } else {
        return maybeOverflow
      }
    }
    default:
      return ColumnWidth
  }
}

export const getSectionMobilePadding = (section?: SectionData) => {
  const type = section && section.type
  const isFillWidth = ["video", "image_collection", "image_set"].includes(type)
  const isMiniImageSet = type === "image_set" && section.layout === "mini"

  return !isFillWidth || isMiniImageSet ? "0 20px" : 0
}
