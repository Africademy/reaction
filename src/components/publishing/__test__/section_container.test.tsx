import * as React from "react"
import * as renderer from "react-test-renderer"
import SectionContainer from "../sections/section_container"
import TextContainer from "../sections/text_container"

it("renders overflow_fillwidth properly", () => {
  const section = {
    layout: "overflow_fillwidth",
    body: "<p>Hello, world!</p>",
  }
  const sectionContainer = renderer
    .create(
      <SectionContainer layout={section.layout}>
        <TextContainer html={section.body} layout="standard" />
      </SectionContainer>
    )
    .toJSON()
  expect(sectionContainer).toMatchSnapshot()
})

it("renders fillwidth properly", () => {
  const section = {
    layout: "fillwidth",
    body: "<p>Hello, world!</p>",
  }
  const sectionContainer = renderer
    .create(
      <SectionContainer layout={section.layout}>
        <TextContainer html={section.body} layout="standard" />
      </SectionContainer>
    )
    .toJSON()
  expect(sectionContainer).toMatchSnapshot()
})

it("renders column_width properly", () => {
  const section = {
    layout: "column_width",
    body: "<p>Hello, world!</p>",
  }
  const sectionContainer = renderer
    .create(
      <SectionContainer layout={section.layout}>
        <TextContainer html={section.body} layout="standard" />
      </SectionContainer>
    )
    .toJSON()
  expect(sectionContainer).toMatchSnapshot()
})
