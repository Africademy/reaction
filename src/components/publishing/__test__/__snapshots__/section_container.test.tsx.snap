// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders column_width properly 1`] = `
<div
  className="sc-bdVaJa iOpAjh"
>
  <div
    className="sc-bwzfXH jTXFN"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hello, world!</p>",
        }
      }
    />
  </div>
</div>
`;

exports[`renders fillwidth properly 1`] = `
<div
  className="sc-bdVaJa eCEkWF"
>
  <div
    className="sc-bwzfXH jTXFN"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hello, world!</p>",
        }
      }
    />
  </div>
</div>
`;

exports[`renders overflow_fillwidth properly 1`] = `
<div
  className="sc-bdVaJa dGifUw"
>
  <div
    className="sc-bwzfXH jTXFN"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hello, world!</p>",
        }
      }
    />
  </div>
</div>
`;
